aztec_gddt/params.py

```diff
@@ -80,14 +85,130 @@
+
+def steady_state_l1_gas_estimate(state: AztecModelState):
+    if state['timestep'] < len(steady_gas_fee_l1_time_series):
+        return steady_gas_fee_l1_time_series[state['timestep']]
+    else:
+        return steady_gas_fee_l1_time_series[-1]
+
+def steady_state_blob_gas_estimate(state: AztecModelState):
+    if state['timestep'] < len(steady_gas_fee_blob_time_series):
+        return steady_gas_fee_blob_time_series[state['timestep']]
+    else:
+        return steady_gas_fee_blob_time_series[-1]
+
+
+STEADY_STATE_GAS_ESTIMATORS = L1GasEstimators(
+    proposal= steady_state_l1_gas_estimate,
+    commitment_bond= steady_state_l1_gas_estimate,
+    content_reveal= steady_state_l1_gas_estimate,
+    content_reveal_blob= steady_state_blob_gas_estimate, # NOTE: this is a HACK assumption, gas was estimated from various documents by Aztec Labs
+    rollup_proof= steady_state_l1_gas_estimate
 )
 
+#############################################################
+## End: Steady state gas estimators defined                ##
+#############################################################
+
+#############################################################
+## Begin: single shock gas estimators defined              ##
+#############################################################
+
+L1_SHOCK_AMOUNT = 100
+BLOB_SHOCK_AMOUNT = 100
+initial_time = floor(0.25 * TIMESTEPS) # XXX: 25% of timesteps
+final_time = floor(0.25 * TIMESTEPS) # XXX: 25% of timesteps
+
+
+single_shock_gas_fee_l1_time_series = np.zeros(TIMESTEPS)
+single_shock_gas_fee_blob_time_series = np.zeros(TIMESTEPS)
+
+single_shock_gas_fee_l1_time_series[0:initial_time] = steady_gas_fee_l1_time_series[0:initial_time].copy()
+single_shock_gas_fee_l1_time_series[-final_time:] = steady_gas_fee_l1_time_series[-final_time:].copy()
+single_shock_gas_fee_l1_time_series[initial_time:TIMESTEPS - final_time] = steady_gas_fee_l1_time_series[initial_time:TIMESTEPS - final_time].copy() + L1_SHOCK_AMOUNT
+
+single_shock_gas_fee_blob_time_series[0:initial_time] = steady_gas_fee_blob_time_series[0:initial_time].copy()
+single_shock_gas_fee_blob_time_series[-final_time:] = steady_gas_fee_blob_time_series[-final_time:].copy()
+single_shock_gas_fee_blob_time_series[initial_time:TIMESTEPS - final_time] = steady_gas_fee_blob_time_series[initial_time:TIMESTEPS - final_time].copy() + L1_SHOCK_AMOUNT
+
+
+def single_shock_l1_gas_estimate(state: AztecModelState):
+    if state['timestep'] < len(single_shock_gas_fee_l1_time_series):
+        return single_shock_gas_fee_l1_time_series[state['timestep']]
+    else:
+        return single_shock_gas_fee_l1_time_series[-1]
+
+def single_shock_blob_gas_estimate(state: AztecModelState):
+    if state['timestep'] < len(single_shock_gas_fee_blob_time_series):
+        return single_shock_gas_fee_blob_time_series[state['timestep']]
+    else:
+        return single_shock_gas_fee_blob_time_series[-1]
+
+SINGLE_SHOCK_GAS_ESTIMATORS = L1GasEstimators(
```

Comment:
note: the above estimators are relevant for the Gas Fee, not for the Gas Units per Tx